packages/rest/src/lib/handlers/Shared.ts

```diff
@@ -65,7 +63,7 @@ export async function makeNetworkRequest(
 	retries: number,
 ) {
 	const controller = new AbortController();
-	const timeout = setTimeout(() => controller.abort(), manager.options.timeout).unref();
+	const timeout = setTimeout(() => controller.abort(), manager.options.timeout);
```

Comment:
```suggestion
	const timeout = setTimeout(() => controller.abort(), manager.options.timeout);
	if (typeof timeout !== 'number') timeout.unref()
```

that way the current behavior stays the same for node users?

Reply:
It's effectively a no-op 
https://github.com/discordjs/discord.js/pull/9682#discussion_r1252393899